### ISSUE TITLE
Update base services implementation

### ISSUE THREAD

**Issue by EuanScott** [EuanScott/project-tweety#10]
## What is it that I can improve upon
Take the already existing base impl for the services.dart and look into improving it in the following by adding in (as per the TODO comments):
- Input Validation
- Logging (dev purposes and to sentry.io/Firebase)
- Tests
- Modularization
- Security

Note: Before any work is done, determine if this is really necessary and if not, just leave out the sections that aren't required.

## Reason for the change
Let's improve on what is already implemented

## Acceptance Criteria
[ ] Does the feature still do what its original intention was?
[ ] Have I made the overall system better? Explain.

## Additional Context
The initial impl was entirely done using AI. See if you can keep on doing it this way.

**Comment by EuanScott** [comment]
hmmm... it seems that this may not be possible as I can't just access the raw JSON data on github. Maybe look into another way of doing this?

Firebase functions perhaps?

**Comment by EuanScott** [comment]
This work has been completed and can be found in the `main` branch.

The following has been done as per the issue description:
1. Input validation
2. Tests

Modularization has been deemed as YAGNI at this stage in the project. Should it be required in the future, it can be looked into.

Logging still need to be investigated and implemented (create a follow-up issue for this to investigate a potential solution (ideally freemium).

## Acceptance Criteria
[x] Does the feature still do what its original intention was?
[x] Have I made the overall system better? Explain.
> I believe that I have. I now have a solution that I can "plug and play" that has been tested to ensure that it works. Overall code quality has gone up.

## Additional Context
This was done majority using duck.ai and github copilot as part of an investigation/feasibility study on my part. AI is a tool that we should all be using.
